src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/viewProvider/LSPSemanticTokensFileViewProviderFactory.java

```diff
@@ -0,0 +1,40 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.semanticTokens.viewProvider;
+
+import com.intellij.lang.Language;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.FileViewProvider;
+import com.intellij.psi.FileViewProviderFactory;
+import com.intellij.psi.PsiManager;
+import com.intellij.psi.SingleRootFileViewProvider;
+import com.redhat.devtools.lsp4ij.LanguageServersRegistry;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * A {@link FileViewProviderFactory} for LSP-backed TextMate files where elements are derived dynamically from reported
+ * semantic tokens.
+ */
+public class LSPSemanticTokensFileViewProviderFactory implements FileViewProviderFactory {
+
+    @Override
+    @NotNull
+    public FileViewProvider createFileViewProvider(@NotNull VirtualFile virtualFile,
```

Comment:
And this is the file view provider factory. It uses the new method for checking whether the provided file/language combination is configured for LSP4IJ before creating our file view provider. Otherwise it just uses the standard file view provider that would have been used for the TextMate file.